canarytokens/windows_fake_fs.py

```diff
@@ -0,0 +1,179 @@
+                "children": [
+                    {"name": "Doc A.pdf", "type": "pdf"},
+                    {"name": "Doc B.docx", "type": "docx"},
+                    {"name": "Doc C.xls", "type": "xls"},
+                    {"name": "Doc D.doc", "type": "doc"},
+                ],
+            },
+        ],
+    },
+    {
+        "name": "Testing",
+        "type": "folder",
+        "children": [
+            {"name": "Doc1.pdf", "type": "pdf"},
+            {"name": "Doc2.xlsx", "type": "xls"},
+            {"name": "Doc3.doc", "type": "doc"},
+        ],
+    },
+    {
+        "name": "Deployment",
+        "type": "folder",
+        "children": [
+            {"name": "doc a.pdf", "type": "pdf"},
+            {"name": "doc b.docx", "type": "docx"},
+            {"name": "doc c.xls", "type": "xls"},
+            {"name": "doc d.doc", "type": "doc"},
+        ],
+    },
+    {
+        "name": "Invoicing",
+        "type": "folder",
+        "children": [
+            {"name": "Invoice A.pdf", "type": "pdf"},
+            {"name": "Invoice B.xlsx", "type": "xls"},
+            {"name": "Invoice C.docx", "type": "docx"},
+            {"name": "Invoice D.doc", "type": "doc"},
+        ],
+    },
+]
+
+
+FOLDER_MAP = {
+    "personal_finances": PERSONAL_FINANCES_STRUCTURE,
+    "home_network": HOME_NETWORK_STRUCTURE,
+    "personal_correspondence": PERSONAL_CORRESPONDENCE_STRUCTURE,
+    "photo_archive": PHOTO_ARCHIVE_STRUCTURE,
+    "defense": DEFENSE_STRUCTURE,
+    "med_tech": MED_TECH_STRUCTURE,
+    "network_admin": NETWORK_ADMIN_STRUCTURE,
+    "security_admin": SECURITY_ADMIN_STRUCTURE,
+    "testing": DUMMY_FOLDER_STRUCTURE,
+}
+
+
+def _gen_ts() -> str:
+    """
+    Generates a random timestamp in the near past
+    """
+    now = floor(datetime.now().timestamp())
+    random_hours = random.randint(1, 10000)
+    return str(now - (random_hours * 60 * 60))
+
+
+def _new_item(path: str, is_folder: bool, size: int = 0) -> dict:
+    isdir = "false"
+    if is_folder:
+        isdir = "true"
+    return {"path": path, "isdir": isdir, "size": size, "timestamp": _gen_ts()}
+
+
+def _process_item(item: dict, path: str) -> str:
+    out = []
+    if item["type"] == "folder":
+        out.append(_new_item(path=path + "\\" + item["name"], is_folder=True))
+        for c in item["children"]:
+            out += _process_item(c, path + "\\" + item["name"])
+    else:
+        out.append(
+            _new_item(
+                path=path + "\\" + item["name"],
+                is_folder=False,
+                size=random.randint(1024, 51200),
+            )
+        )
+    return out
+
+
+def _json_to_csv(fake_file_structure: list[dict]) -> str:
+    """
+    Converts a JSON-based dict of a file system structure to the CSV format used by the ProjFS provider
+    """
+
+    fieldnames = ["path", "isdir", "size", "timestamp"]
+    out = []
+    for item in fake_file_structure:
+        out += _process_item(item, "")
```

Comment:
```suggestion
    out = [_process_item(item, "") for item in fake_file_structure]
```

More concise, but feel free to ignore

Reply:
I cleaned it up using .extend but a single line generator style for won't work for same reason as above